### PR TITLE
Add rand(Interval)

### PR DESCRIPTION
**Pull request by schillic**
This is a proposal of something I find useful. There was a related discussion in #109.

### CODE REVIEW

File: src/rand.jl [hunk 1]
```diff
@@ -8,6 +9,17 @@ Base.rand(X::IntervalBox) = rand.(X)
 Random.gentype(::Type{Interval{T}}) where {T} = T
 Random.rand(rng::AbstractRNG, X::Random.SamplerTrivial{Interval{T}}) where {T} = rand(X[])
 
+Base.rand(type::Type{<:Interval}) = rand(GLOBAL_RNG, type)
+Base.rand(::Type{Interval}) = rand(Interval{Float64})
+
+function Random.rand(rng::AbstractRNG, ::Type{Interval{T}}) where {T}
```
Comment (lucaferranti) [inline src/rand.jl:15]: should this be `Random.randn` if it uses randn internally?

Comment (schillic) [reply src/rand.jl:15]: I guess so, yes.

Comment (schillic) [reply src/rand.jl]: I changed the name.

**Comment by schillic** [comment]
Any opinion on this PR?

**Comment by Kolaru** [comment]
I personally don't really like it. As far as I know, there is no accepted definition of what a random interval should be, let alone a normally distributed one.

My expectation would be that anyone needing a random interval would need something slightly different. Since it is easy enough (`interval(sort(randn(2))...)` in the present case), and unless I am missing something, I am rather in favor of not merging it.

**Pull request closed without merging**